src/datapilot/core/platforms/dbt/insights/checks/check_model_has_tests_by_type.py

```diff
@@ -0,0 +1,69 @@
+from typing import List
+
+from datapilot.config.utils import get_test_type_configuration
+from datapilot.core.insights.utils import get_severity
+from datapilot.core.platforms.dbt.insights.checks.base import ChecksInsight
+from datapilot.core.platforms.dbt.insights.schema import DBTInsightResult
+from datapilot.core.platforms.dbt.insights.schema import DBTModelInsightResponse
+from datapilot.core.platforms.dbt.schemas.manifest import AltimateResourceType
+
+
+class CheckModelHasTestsByType(ChecksInsight):
+    NAME = "Check Model Has Tests By Type"
```

Comment:
et a number of key-value pairs."
        " Key is type of test (data or schema) and value is required "
        "minimal number of tests eg. --test data=1 schema=2"
        "(do not put spaces before or after the = sign). "
        "",

Reply:
we are just checking if the test by type is there or not. we are getting that from the config. they are getting the count as well. if the type of test is more than 1